### PR TITLE
Provide documentation for the pulp-cli plugin

### PR DESCRIPTION
**Pull request by mikalstill**
It would be nice to not have to make raw HTTP calls to manage repositories. This PR at the moment is just a minimal implementation of creating a repository, but I wanted to ask how you'd feel about a CLI plugin.

### CODE REVIEW

**Comment by mdellweg** [comment]
https://github.com/pulp/pulp-cli-deb

Please don't duplicate the work. ;)
Alas, it's documented there: https://docs.pulpproject.org/pulp_cli/#cli-plugins

**Comment by mikalstill** [comment]
I didn't even realise this existed. May I ask why the `pulp-deb` docs use `http` instead of this CLI?

**Comment by quba42** [comment]
> I didn't even realise this existed. May I ask why the `pulp-deb` docs use `http` instead of this CLI?

Because pulp-cli-deb did not exist at the time the docs were written (and also pulp-cli-deb is not as fully feature full as the CLI for some of the other plugins). Adding docs about pulp-cli-deb is definitely wanted, but nobody has time to work on them. :disappointed:

**Comment by quba42** [comment]
I guess it might be worth making it a priority to add at least some minimal docs to inform users that pulp-cli-deb exists.

**Comment by mikalstill** [comment]
Thanks for the reply. I'm happy enough to pivot this PR into being tweaks to the docs pointing at the exist CLI plugin. Do I need to create an issue for that before updating this PR?

**Comment by quba42** [comment]
> Do I need to create an issue for that before updating this PR?

I went ahead and created one you can use: https://github.com/pulp/pulp_deb/issues/598

**Comment by mikalstill** [comment]
Questions and comments from a first pass at this:

* is it just me, or is it a bit clumbersome that the CLI plugin returns JSON blocks that a user then needs to parse?
* there are obvious gaps for sync / upload in the CLI plugin.

**Comment by quba42** [comment]
> * is it just me, or is it a bit clumbersome that the CLI plugin returns JSON blocks that a user then needs to parse?

AFAIK this is exactly the same as for Pulp CLI for any other Pulp plugin like pulp_rpm, though I believe it is also configurable:
```
# pulp config create --help
  --format [json|yaml|none]       Format of the response
```
Never tried any of the other formats, so I have no idea if they are more or less convenient.

> * there are obvious gaps for sync / upload in the CLI plugin.

I am not aware of any gaps for sync, but there are certainly known feature gaps including for upload: https://github.com/pulp/pulp-cli-deb/issues

File: docs/pulp_cli_deb.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+line tool available.
+
+To install the command line plugin, simply `pip install` it into the python
+environment that your `pulp-cli` is installed into. This is most likely the
+system pip if you have used the `pulp-installer` ansible based installation
+method. In that case, you can install the CLI plugin like this:
+
+.. code-block:: bash
+
+    sudo pip3 install pulp_cli_deb
```
Comment (mdellweg) [inline docs/pulp_cli_deb.rst:17]: Please do not use `sudo pip`. Talk about a virtual env if you like...

**Comment by quba42** [comment]
Note that we will now be adding Pulp CLI workflows for the pulp_deb 3.0 release: https://github.com/pulp/pulp_deb/pull/876

**Pull request closed without merging**